presto-main/src/main/java/com/facebook/presto/type/ArrayParametricType.java

```diff
@@ -40,7 +40,9 @@ public String getName()
     @Override
     public Type createType(List<TypeParameter> parameters)
     {
-        checkArgument(parameters.size() == 1, "Array type expects exactly one type as a parameter, got %s", parameters);
+        if (parameters.isEmpty()) {
```

Comment:
To do that we need a signature for parametric types (on a side note, what do you think of renaming them TypeConstructor?), do you think that's something we could put in the Teradata refactoring?